tests/manual/text/EmojiTest.java

```diff
@@ -23,64 +23,42 @@
  * questions.
  */
 
-import javafx.application.Application;
-import javafx.application.Platform;
-import javafx.scene.layout.HBox;
-import javafx.scene.layout.VBox;
-import javafx.scene.Group;
-import javafx.scene.Scene;
-import javafx.scene.control.Button;
+import javafx.scene.Node;
 import javafx.scene.control.Label;
 import javafx.scene.control.TextField;
-import javafx.stage.Stage;
+import javafx.scene.layout.VBox;
 import javafx.scene.text.Font;
 import javafx.scene.text.Text;
+import com.oracle.util.testing.ManualTestWindow;
 
-public class EmojiTest extends Application {
-
-   static String instructions =
-      """
-        This tests rendering of Emoji glyphs which is only supported on macOS.
-        On macOS you should see a yellow-coloured smiling face image,
-        embedded between 'ab' and 'cd'.
-        On other platforms it may be a missing glyph, or an empty space, or
-        a similar rendering as a greyscale/B&W glyph.
-        Principally, you are checking that the emoji is rendered on macOS in
-        each of the controls and nodes displayed in the test, and that the
-        editable text field handles selection of the emoji glyph with the
-        same background as other glyphs - this presumes the emoji image has
-        transparent background pixels.
-        There are 3 different ways it is displayed to verify
-        1) Text node. 2) Label control, 3) TextField Control
-        Press the Pass or Fail button as appropriate and the test will exit.
-        If what you see is not explained here, ask before filing a bug.
-
-
-        """;
-
+public class EmojiTest {
```

Comment:
Everything, including creating an instance of Application, is taken care of by the `ManualTestWindow`.  This simplifies the test code greatly.

What is the reason that the test must extend `Application`?

Reply:
For example if test needs to access `getParameters()` or `getHostServices ()` from `Application` how it can be done? I think it might be better to extend `ManualTestWindow` from `Application` and require test itself to extend `ManualTestWindow`. In this case tests will extend `Application`.